Sources/NetworkProtection/KeyManagement/NetworkProtectionKeyStore.swift

```diff
@@ -25,19 +25,26 @@ public protocol NetworkProtectionKeyStore {
     ///
     func currentKeyPair() -> KeyPair
 
-    /// Sets the validity interval for keys
+    /// Create a new `KeyPair`.
     ///
+    func newKeyPair() -> KeyPair
 
+    /// Sets the validity interval for keys
+    ///
     func setValidityInterval(_ validityInterval: TimeInterval?)
 
+    /// Updates the existing KeyPair.
+    ///
+    func updateKeyPair(_ newKeyPair: KeyPair)
+
     /// Updates the current `KeyPair` to have the specified expiration date
     ///
     /// - Parameters:
-    ///     - newExpirationDate: the new expiration date for the keypair
+    ///     - newExpirationDate: the new expiration date for the KeyPair
     ///
-    /// - Returns: a new keypair with the specified updates
+    /// - Returns: a new KeyPair with the specified updates
     ///
-    func updateCurrentKeyPair(newExpirationDate: Date) -> KeyPair
+    func updateKeyPairExpirationDate(_ newDate: Date) -> KeyPair
```

Comment:
Renamed to make sure this is easy to distinguish from the new `updateKeyPair` function.